cmd/oras/internal/display/status/discard.go

```diff
@@ -45,3 +45,11 @@ func (DiscardHandler) TrackTarget(gt oras.GraphTarget) (oras.GraphTarget, error)
 
 // UpdateCopyOptions updates the copy options for the artifact push.
 func (DiscardHandler) UpdateCopyOptions(opts *oras.CopyGraphOptions, fetcher content.Fetcher) {}
+
+// OnTagged implements TagHandler interface.
+func (DiscardHandler) OnTagged(tag string) error {
+	return nil
+}
+
+// PreTagging implements TagHandler interface.
+func (DiscardHandler) PreTagging(reference string) {}
```

Comment:
I know these is all implemented this way already, but green field, I would think of text as a format and we wouldn't be discarding anything, it would just be another format.

Reply:
First of all, it's necessary to do so since user explicitly wants to make the output structured via `--format <json | go-template>`, that's why we need to add such implementation to discard text output.

Secondly, you are right about making the tagging info into another format here. We should add the tag metadata into the structured output of `oras push`. The tag metadata currently not included in [spec PR](https://github.com/oras-project/oras/pull/1199) because it's user input and there is no clear use case to use it to automate stuffs.